### ISSUE TITLE
False positives indent errors after chained methods PSR-2 standard

### ISSUE THREAD

**Issue by hskrasek** [squizlabs/PHP_CodeSniffer#1002]
I ran into an issue where if I chained two methods that both received closures as callbacks, phpcs would falsely claim that the next block was indented too many spaces.

Here is a gist with what my code looks like, and what phpcs claims is right: https://gist.github.com/hskrasek/90cd5cf2873198a56a07fc30ebc3db61

And what phpcs outputs:

```bash
$ phpcs -n --standard=PSR2 app/Modules/TV/src/Models/v1/VideosRepository.php

FILE: ...s/Projects/api/app/Modules/TV/src/Models/v1/VideosRepository.php
----------------------------------------------------------------------
FOUND 2 ERRORS AFFECTING 2 LINES
----------------------------------------------------------------------
 133 | ERROR | [x] Line indented incorrectly; expected 4 spaces,
     |       |     found 8
 135 | ERROR | [x] Line indented incorrectly; expected 4 spaces,
     |       |     found 8
----------------------------------------------------------------------
PHPCBF CAN FIX THE 2 MARKED SNIFF VIOLATIONS AUTOMATICALLY
----------------------------------------------------------------------

Time: 53ms; Memory: 6Mb
```

I'm using version 2.6.0 (stable)

**Comment by hskrasek** [comment]
Anyone else running into or seeing this issue? Or am I somehow the only with that this is happening to?

**Comment by voda** [comment]
I ran into the same issue.  If you put the code outside a class and method, than it expects -4 spaces. With this code:
```php
$var = $collection->map(function($value) {
    return $value;
})->filter(function($value) {
    return $value !== null;
});
if ($var) {
    echo "A";
}
```
the output for command `phpcs --standard=PSR2 -np --sniffs=Generic.WhiteSpace.ScopeIndent --report-width=160 a.php` is
```
-------------------------------------------------------------------------
FOUND 2 ERRORS AFFECTING 2 LINES
-------------------------------------------------------------------------
  8 | ERROR | [x] Line indented incorrectly; expected -4 spaces, found 0
 10 | ERROR | [x] Line indented incorrectly; expected -4 spaces, found 0
-------------------------------------------------------------------------
PHPCBF CAN FIX THE 2 MARKED SNIFF VIOLATIONS AUTOMATICALLY
-------------------------------------------------------------------------
```
Tried looking into the code, but don't even know where to start looking.

**Comment by hskrasek** [comment]
@voda Just glad someone else is running into this, hopefully someone from the team sees this and can help out soon. I'm gonna try and look into the codebase as well

**Comment by hskrasek** [comment]
So some digging seems to reveal that the tokens list shows type as `T_IF` but the code as 304, which if you look into the constants defined by the tokenizer, 304 is the value assigned to `T_ELSE`. I'm thinking that might be causing the issue

**Comment by hskrasek** [comment]
Interestingly enough I made a tweak to my code and the error goes away. 

When I changed the code from what I have in the gist to what I have below, the error went away.

```php
public function method() {
    $videos = (new Collection($videos))
        ->transform(function ($video) {
            return (new Video)->hydrate($video);
        })->filter(function ($video) {
            return $video->enabled;
        });

    if ($videos->isEmpty()) {
        throw new VideoDisabledException;
    }
}
```

**Comment by gsherwood** [comment]
Turns out this is a duplicate of #961
